src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_COUNT = 5;
+    private static final int MAX_VALUE_OF_PARAMETER = 100;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+
+        Figure figure;
+
+        switch (figureNumber) {
+            case 0:
+                figure = new Square(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_VALUE_OF_PARAMETER));
+                break;
```

Comment:
```suggestion
            case 0:
                return getRandomSquare();
```
lets add private methods to get random figures - this would make your code easier to read